workloads/virtual-machines/unresponsive-nodes.md

```diff
@@ -0,0 +1,89 @@
+# Detecting and resolving Node issues
+
+KubeVirt has its own node daemon, called `virt-handler`. In addition to the
+usual node issue detection in k8s, our daemon has its own heartbeat mechanism
+too, in order to allow us fine-tuned error handling for our
+`VirtualMachineInstances`.
+
+## Heartbeat of virt-handler
+
+`virt-handler` periodically tries to update the `kubevirt.io/schedulable` label
+and the `kubevirt.io/heartbeat` annotation on the node it is running on:
+
+```bash
+$ kubectl get nodes -o yaml
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Node
+  metadata:
+    annotations:
+      kubevirt.io/heartbeat: 2018-11-05T09:42:25Z
+    creationTimestamp: 2018-11-05T08:55:53Z
+    labels:
+      beta.kubernetes.io/arch: amd64
+      beta.kubernetes.io/os: linux
+      cpumanager: "false"
+      kubernetes.io/hostname: node01
+      kubevirt.io/schedulable: "true"
+      node-role.kubernetes.io/master: ""
+```
+
+If a `VirtualMachineInstance` gets scheduled, the scheduler is only considering
+nodes where `kubevirt.io/schedulable` is `true`. This can be seen when looking
+on the corresponding pod of a `VirtualMachineInstance`:
+
+```bash
+$ kubectl get pods  virt-launcher-vmi-nocloud-ct6mr -o yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  [...]
+spec:
+  [...]
+  nodeName: node01
+  nodeSelector:
+    kubevirt.io/schedulable: "true"
+  [...]
+```
+
+In case there is a communication issue or the host goes down, `virt-handler`
+can't update its labels and annotations any-more. Once the last
+`kubevirt.io/heartbeat` timestamp is older than five minutes, the KubeVirt
+node-controller kicks in and sets the `kubevirt.io/schedulable` label to
+`false`. As a consequence no VMIs will be scheduled any-more to this node, until
```

Comment:
Suggest rewording to:

As a consequence no more VMIs will be schedule to this node until `virt-handler` is connected again.